include/dirac_quda.h

```diff
@@ -205,6 +205,21 @@ namespace quda {
     */
     virtual bool isCoarse() const { return false; }
 
+    /**
+      @brief return if the operator is a Wilson-type 4-d operator
+     */
+    virtual bool isWilsonType() const = 0;
```

Comment:
Having every operator need to specify this seems a bit cumbersome.  Why not default all of these operators to return false, and then specialize where needed?  `DiracWilson` can return `true` and all children will then inherit that.  Similarly for staggered and DWF.

Reply:
It was cumbersome for me, but it's a marginal effort for anyone going forward. The reason why I made it abstract and not have it return false by default is to force folks to think about it, but again only once.

I do like the idea of setting it to true for sufficiently high-up "base" classes such as `DiracWilson`, so I'll pop that in.

Reply:
https://github.com/lattice/quda/pull/1488/commits/b4b9e91ad6c2a222f440795e7c1db2212c6a7aa5